engines/cryo/eden.cpp

```diff
@@ -0,0 +1,9588 @@
+	if (_globals->_curObjectId != 0 && _globals->_curObjectId < Objects::obTablet1)
+		return;
+	_globals->_curObjectId = 0;
+	getObject(Objects::obTablet2);
+	putObject();
+	for (int i = 0; i < 6; i++)
+		_objects[Objects::obTablet1 - 1 + i]._count = 0;
+	_globals->_curObjectFlags = 0;
+	_globals->_inventoryScrollPos = 0;
+	_globals->_curObjectCursor = 9;
+	_gameIcons[16]._cursorId |= 0x8000;
+	showObjects();
+	_gameRooms[131]._video = 0;
+	hideBars();
+	playHNM(149);
+	_globals->_varF1 = RoomFlags::rf04;
+	_globals->_drawFlags = DrawFlags::drDrawFlag20;
+	_normalCursor = true;
+	maj2();
+}
+
+// Original name: voirlac
+void EdenGame::actionLookLake() {
+	perso_t *perso = &kPersons[PER_MORKUS];
+	Room *room = _globals->_roomPtr;
+	Area *area = _globals->_areaPtr;
+	int16 vid = _globals->_curObjectId == Objects::obApple ? 81 : 54;
+	for (++perso; perso->_roomNum != 0xFFFF; perso++) {
+		if (perso->_roomNum != _globals->_roomNum)
+			continue;
+		vid++;
+		if (_globals->_curObjectId != Objects::obApple)
+			continue;                   //TODO: pc breaks here
+		if ((perso->_flags & PersonFlags::pfTypeMask) != PersonFlags::pftMosasaurus)
+			continue;
+		if (!(perso->_flags & PersonFlags::pf80))
+			return;
+		perso->_flags &= ~PersonFlags::pf80; //TODO: useless? see above
+		area->_flags |= AreaFlags::afFlag8;
+		_globals->_curAreaFlags |= AreaFlags::afFlag8;
+		room->_id = 3;
+	}
+	debug("sea monster: room = %X, d0 = %X\n", _globals->_roomNum, _globals->_roomImgBank);
+	hideBars();
+	playHNM(vid);
+	updateRoom(_globals->_roomNum);           //TODO: getting memory trashed here?
+	if (_globals->_curObjectId == Objects::obApple)
+		loseObject(Objects::obApple);
+	_globals->_eventType = EventType::etEventF;
+	showEvents();
+}
+
+// Original name: gotohall
+void EdenGame::actionGotoHall() {
+	_globals->_prevLocation = _globals->_roomNum & 0xFF;
+	deplaval((_globals->_roomNum & 0xFF00) | 6);
+}
+
+// Original name: demitourlabi
+void EdenGame::actionLabyrinthTurnAround() {
+	_globals->_prevLocation = _globals->_roomNum & 0xFF;
+	_globals->_var100 = 0xFF;
+	uint16 target = (_globals->_roomNum & 0xFF00) | _globals->_roomPtr->_exits[2];
+	faire_suivre(target);
+	_globals->_roomNum = target;
+	_globals->_eventType = EventType::etEvent5;
+	updateRoom(_globals->_roomNum);
+}
+
+// Original name: gotonido
+void EdenGame::actionGotoFullNest() {
+	_globals->_roomPtr++;
+	_globals->_eventType = 0;
+	_globals->_roomImgBank = _globals->_roomPtr->_bank;
+	_globals->_roomVidNum = _globals->_roomPtr->_video;
+	_globals->_curRoomFlags = _globals->_roomPtr->_flags;
+	_globals->_varF1 = _globals->_roomPtr->_flags;
+	animpiece();
+	_globals->_var100 = 0;
+	maj2();
+}
+
+// Original name: gotoval
+void EdenGame::actionGotoVal() {
+	uint16 target = _globals->_roomNum;
+	rundcurs();
+	display();
+	_scrollPos = 0;
+	char obj = _curSpot2->_objectId - 14;    //TODO
+	_globals->_prevLocation = target & 0xFF;
+	deplaval((target & 0xFF00) | obj);  //TODO careful!
```

Comment:
Again, I don't understand these TODOs... what needs to be done here, and why should we be careful?